doc/hacking/snmp-tunnels.rst

```diff
@@ -16,6 +16,8 @@ combination of SSH and socat.
 
 This guide will document a way to establish such an SNMP tunnel.
 
+It does not work if you are on a NATted network.
```

Comment:
That was not the point of my comments in my previous review. Tunneling will still work through a NATted network.

*Specifically, using `curl ifconfig.me` to get your own IP address does not work when you're on a NATted network.*

My point was that in many cases, you won't care about your public IP address, since you will be running NAV in development on your localhost, and setting up the tunnel from your localhost, so using 127.0.0.1 is fine.

However, things get slightly more complicated if you add Docker in the mix, by using the docker-compose dev environment, which we usually do in the NAV team.

If your host OS is Linux, you can get away with using the internal router address your docker containers are talking through:

Example from a `nav` container within docker-compose:

```console
root@ca5bc4f41655:/# ip route
default via 172.27.0.1 dev eth0
172.27.0.0/16 dev eth0 proto kernel scope link src 172.27.0.4
```

My host can be reached as `172.27.0.1` here.

However, in Docker on MacOS (and likely Windows too) it's not that simple, and you're better off using your host's address on the LAN you're connected to. How to find your LAN IP without talking to an external service varies wildly between operating systems and your hardware.

Specifically, on a Mac connected to a wireless network, you can ⌥+click on the WiFi-symbol on the top menu line, and a menu with connection details, including your IP address, should appear.



Other options include establishing the tunnel from within the docker container you will be using it from. In that case, you can still use 127.0.0.1. Or, you could add an extra container to your docker-compose network whose sole purpose is to act as an SNMP forwarder. The latter could be extra useful if you want to forward traffic to multiple SNMP agents. You could define a service for each of them in `docker-compose.override.yml`.

Reply:
Come to think of it: My last idea for this has actually been sort of implemented but not completely documented in NAV: In `docker-compose.yml` you will find a commented-out service entry that sets up SNMP forwarding using SSH and socat as a service container, just as I described:

https://github.com/Uninett/nav/blob/16bb0c59ff79a43c816c539a86e76f89cc77c3b6/docker-compose.yml#L105-L107

This is even facilitated by a script provided in NAV: 

https://github.com/Uninett/nav/blob/master/tools/forward/snmp_forward.sh

You  might actually want to test out this latter approach and include that in the docs too :)